src/apimgmt/azext_apimgmt/_params.py

```diff
@@ -0,0 +1,1459 @@
+        c.argument('_type', options_list=['--type'], arg_type=get_enum_type(['http', 'soap']), id_part=None, help='Type of API.')
+        c.argument('api_revision', id_part=None, help='Describes the Revision of the Api. If no value is provided, default revision 1 is created')
+        c.argument('api_version', id_part=None, help='Indicates the Version identifier of the API if the API is versioned')
+        c.argument('is_current', arg_type=get_three_state_flag(), id_part=None, help='Indicates if API revision is current api revision.')
+        c.argument('api_revision_description', id_part=None, help='Description of the Api Revision.')
+        c.argument('api_version_description', id_part=None, help='Description of the Api Version.')
+        c.argument('api_version_set_id', id_part=None, help='A resource identifier for the related ApiVersionSet.')
+        c.argument('subscription_required', arg_type=get_three_state_flag(), id_part=None, help='Specifies whether an API or Product subscription is required for accessing the API.')
+        c.argument('source_api_id', id_part=None, help='API identifier of the source API.')
+        c.argument('display_name', id_part=None, help='API name. Must be 1 to 300 characters long.')
+        c.argument('service_url', id_part=None, help='Absolute URL of the backend service implementing this API. Cannot be more than 2000 characters long.')
+        c.argument('path', id_part=None, help='Relative URL uniquely identifying this API and all of its resource paths within the API Management service instance. It is appended to the API endpoint base URL specified during the service instance creation to form a public URL for this API.')
+        c.argument('protocols', id_part=None, help='Describes on which protocols the operations in this API can be invoked.')
+        c.argument('api_version_set', id_part=None, help='Version set details')
+        c.argument('value', id_part=None, help='Content value when Importing an API.')
+        c.argument('_format', options_list=['--format'], arg_type=get_enum_type(['wadl-xml', 'wadl-link-json', 'swagger-json', 'swagger-link-json', 'wsdl', 'wsdl-link', 'openapi', 'openapi+json', 'openapi-link']), id_part=None, help='Format of the Content in which the API is getting imported.')
+        c.argument('wsdl_selector', id_part=None, help='Criteria to limit import of WSDL to a subset of the document.')
+        c.argument('api_type', arg_type=get_enum_type(['SoapToRest', 'SoapPassThrough']), id_part=None, help='Type of Api to create. <br> * `http` creates a SOAP to REST API <br> * `soap` creates a SOAP pass-through API .')
+
+    with self.argument_context('apimgmt api delete') as c:
+        c.argument('resource_group', id_part=None, help='The name of the resource group.')
+        c.argument('service_name', id_part=None, help='The name of the API Management service.')
```

Comment:
this should be aliased to `--name` & `-n` and id_part should be the `name`. Again consistency is very important. I would also suggest you can handle the grand child resource if there are